addons/project_purchase/static/src/product_catalog/kanban_record.js

```diff
@@ -0,0 +1,16 @@
+/** @odoo-module **/
+import { patch } from "@web/core/utils/patch";
+
+import { ProductCatalogKanbanRecord } from "@product/product_catalog/kanban_record";
+
+patch(ProductCatalogKanbanRecord.prototype, {
+    _getUpdateQuantityAndGetPrice() {
+        return {
+            order_id: this.env.orderId,
+            product_id: this.env.productId,
+            quantity: this.productCatalogData.quantity,
+            res_model: this.env.orderResModel,
+            project_id: this.props.record.context.project_id,
+        };
```

Comment:
Shouldn't something like this works ?
```suggestion
        return {
            ...super._getUpdateQuantityAndGetPrice(),
            project_id: this.props.record.context.project_id,
        };
```